use-timescale/compression/compression-design.md

```diff
@@ -0,0 +1,135 @@
+
+|Timestamp|Device ID|Status Code|Temperature|
+|-|-|-|-|
+|[12:00:01, 12:00:01, 12:00:02, 12:00:02, 12:00:03, 12:00:03]|[A, B, A, B, A, B]|[0, 0, 0, 0, 0, 4]|[70.11, 69.70, 70.12, 69.69, 70.14, 69.70]|
+
+Even before you compress any data, this format immediately saves storage by
+reducing the per-row overhead. PostgreSQL typically adds around 27 bytes of
+overhead per row. So even without any compression, if our schema above is say 32
+bytes, then 1000 rows of data which previously took about 59 kilobytes
+(`1000 x (32 + 27) ~= 59 kilobytes`), now takes about 32 kilobytes
+(`1000 x 32 + 27 ~= 32 kilobytes`) in this format.
+
+This format arranges the data so that similar data, such as timestamps, device
+IDs, or temperature readings, is stored contiguously. This means that you can
+then use type-specific compression algorithms to compress the data further, and
+each array is separately compressed. For more information about the compression
+methods used, see the [compression methods section][compression-methods].
+
+When the data is in array format, you can perform queries that require a subset
+of the columns very quickly. For example, if you have a query like this one, that
+asks for the average temperature over the past day:
+
+<CodeBlock canCopy={false} showLineNumbers={false} children={`
+SELECT time_bucket(‘1 minute’, timestamp) as minute
+ AVG(temperature)
+FROM table
+WHERE timestamp > now() - interval ‘1 day’
+ORDER BY minute DESC
+GROUP BY minute;
+`} />
+
+The query engine can fetch and decompress only the timestamp and temperature
+columns to efficiently compute and return these results.
+
+Finally, TimescaleDB uses non-inline disk pages to store the compressed arrays.
+This means that the in-row data points to a secondary disk page that stores the
+compressed array, and the actual row in the main table becomes very small,
+because it is now just pointers to the data. When data stored like this is
+queried, only the compressed arrays for the required columns are read from disk,
+further improving performance by reducing disk reads and writes.
+
+## Querying compressed data
+
+In the previous example, the database has no way of knowing which rows need to
+be fetched and decompressed to resolve a query. For example, the database can't
+easily determine which rows contain data from the past day, as the timestamp
+itself is in a compressed column. You don't want to have to decompress all the
+data in a chunk, or even an entire hypertable, to determine which rows are
+required.
+
+TimescaleDB automatically includes more information in the row and includes
+additional groupings to improve query performance. When you compress a
+hypertable, either manually or through a compression policy, you need to specify
+an `ORDER BY` column.
+
+`ORDER BY` columns specify how the rows that are part of a compressed patch are
+ordered. For most time-series workloads, this is by timestamp, but you can also
+specify a second dimension, such as location.
```

Comment:
You can specify as many columns as you like, even though it might not be very efficient. This sounds like you can only add one extra column.